R/getExperimentCrossCorrelation.R

```diff
@@ -0,0 +1,577 @@
+        xvec <- xvec[keep]
+        yvec <- yvec[keep]
+        
+        # Number of data-annotation samples for
+        # calculating the correlations
+        n <- sum(keep)
+        # Calculate cross-correlation using Goorma and Kruskal tau
+        correlations[varname, lev] <- .calculate_gktau(xvec, yvec) 
+      }
+    }
+  }
+  
+  # If there are p_values that are not NA, adjust them
+  if (!all(is.na(p_values))) {
+    # Corrected p-values
+    p_values_adjusted <- array(NA, dim=dim(p_values))
+    p_values_adjusted <- matrix(p.adjust(p_values, method=p_adj_method), nrow=nrow(p_values))
+    dimnames(p_values_adjusted) <- dimnames(p_values)
+  }
+  
+  return(list(correlations = correlations, 
+              p_values = p_values, 
+              p_values_adjusted = p_values_adjusted))
+}
+
+############################## .correlation_filter #############################
+.correlation_filter <- function(correlations,
+                                p_values,
+                                p_values_adjusted,
+                                p_adj_threshold,
+                                n_signif,
+                                cor_threshold, 
+                                sort,
+                                assay1, assay2, filter_self_correlations){
+  # Filter
+  if (!is.null(p_adj_threshold) || !is.null(cor_threshold)) {
+    
+    # Replace NAs with extreme values for filtering purposes
+    p_values_adjusted[is.na(p_values_adjusted)] <- 1
+    p_values[is.na(p_values_adjusted)] <- 1
+    correlations[is.na(correlations)] <- 0
+    
+    # Filter by adjusted p-values and correlations
+    inds1.q <- inds2.q <- inds1.c <- inds2.c <- NULL
+    # Which features have more adjusted p-values under the threshold than the 
+    # 'n_signif' specifies
+    if (!is.null(p_adj_threshold)) {
+      inds1.q <- apply(p_values_adjusted, 1, function(x) {
+        sum(x < p_adj_threshold) >= n_signif
+      })
+      inds2.q <- apply(p_values_adjusted, 2, function(x) {
+        sum(x < p_adj_threshold) >= n_signif
+      })
+    }
+    # Which features have correlation over correlation threshold?
+    if (!is.null(cor_threshold)) {
+      inds1.c <- apply(abs(correlations), 1, function(x) {
+        sum(x > cor_threshold | x < cor_threshold) >= n_signif
+      })
+      inds2.c <- apply(abs(correlations), 2, function(x) {
+        sum(x > cor_threshold | x < cor_threshold) >= n_signif
+      })
+    }
+    # Combine results from previous steps
+    if (!is.null(p_adj_threshold) && !is.null(cor_threshold)) {
+      inds1 <- inds1.q & inds1.c
+      inds2 <- inds2.q & inds2.c
+    } else if (is.null(p_adj_threshold) && !is.null(cor_threshold)) {
+      inds1 <- inds1.c
+      inds2 <- inds2.c
+    } else if (!is.null(p_adj_threshold) && is.null(cor_threshold)) {
+      inds1 <- inds1.q
+      inds2 <- inds2.q
+    }
+    
+    # If both features have more TRUEs than n_signif specifies /
```

Comment:
This was feature from `microbiome::associate`

This get the subset of this correlation table --> subset includes only those features that have more significant correlations ( > `n_signif`)

I'm not sure if this is useful, because there can be very strong correlation between individual `feature1` and individual `feature2`, but it can be subsetted off, because of this threshold. 

However, with this it is possible to find features that correlates more holistically with other feature set

I am not sure if we should implement this `n_signif`. This part could be just "get only those features that have at least one significant correlation"

Any ideas @antagomir ?